plugins/draco/io/DracoReader.cpp

```diff
@@ -0,0 +1,363 @@
+    }
+
+    m_pc = std::move(pc_status).value();
+
+    m_count = m_pc->num_points();
+}
+
+
+void DracoReader::addDimensions(PointLayoutPtr layout)
+{
+    using namespace draco;
+
+    log()->get(LogLevel::Debug) << "draco pc num_attributes: "
+                                << m_pc->num_attributes() << std::endl;;
+
+    const GeometryMetadata *metadata = m_pc->GetMetadata();
+    for (int i=0; i < m_pc->num_attributes(); ++i)
+    {
+        const PointAttribute* attr = m_pc->GetAttributeByUniqueId(i);
+        const AttributeMetadata* attr_metadata = m_pc->GetAttributeMetadataByAttributeId(i);
+        DataType dt = attr->data_type();
+        GeometryAttribute::Type at = attr->attribute_type();
+
+        int8_t nc = attr->num_components();
+        log()->get(LogLevel::Debug) << "named component: "
+                                    << GeometryAttribute::TypeToString(at)
+                                    << " subcomponents: " << (int)nc << std::endl;;
+        std::string name;
+        if (attr_metadata)
+        {
+            log()->get(LogLevel::Debug) << "number of metadata entries: "
+                                        << attr_metadata->num_entries()
+                                        << std::endl;
+            attr_metadata->GetEntryString("name", &name);
+            log()->get(LogLevel::Debug) << "metadata name: "
+                                        << name
+                                        << std::endl;
+        }
+        switch (at)
+        {
+            case GeometryAttribute::POSITION:
+            {
+                layout->registerDim(Dimension::Id::X);
+                layout->registerDim(Dimension::Id::Y);
+                layout->registerDim(Dimension::Id::Z);
+                break;
+            }
+            case GeometryAttribute::NORMAL:
+            {
+                layout->registerDim(Dimension::Id::NormalX);
+                layout->registerDim(Dimension::Id::NormalY);
+                layout->registerDim(Dimension::Id::NormalZ);
+                break;
+            }
+            case GeometryAttribute::COLOR:
+            {
+                layout->registerDim(Dimension::Id::Red);
+                layout->registerDim(Dimension::Id::Green);
+                layout->registerDim(Dimension::Id::Blue);
+            }
+            case GeometryAttribute::TEX_COORD:
+            {
+                layout->registerDim(Dimension::Id::TextureU);
+                layout->registerDim(Dimension::Id::TextureV);
+                if (nc == 3)
+                {
+                    m_textureW = true;
+                    layout->registerDim(Dimension::Id::TextureW);
+                }
+                break;
+            }
+
+            case GeometryAttribute::GENERIC:
+            {
+                Dimension::Id id = pdal::Dimension::id(name);
+                if (id != Dimension::Id::Unknown) {
+                    m_generics[id] = std::vector<double>(m_pc->num_points(), 0.0);
```

Comment:
Are generic dimensions always read into double, or can the user control this?

Reply:
User will not be setting this.